### PR TITLE
samples(spanner): update samples to use wrappers

### LINKED ISSUE #891: samples(spanner): update samples to use new client wrapper
### Use the new wrapper instead of initializing the generated client directly.
- [x] spanner_get_instance_config
- [x] spanner_list_instance_configs
- [x] spanner_list_databases
- [x] spanner_create_database_with_default_leader
- [x] spanner_update_database_with_default_leader
- [x] spanner_get_database_ddl
- [x] spanner_add_json_column

### Use the new wrapper instead of managing instances, databases and backups through the Spanner client.
- [x] spanner_create_instance
- [x] spanner_create_instance_with_processing_units
- [x] spanner_create_database_with_encryption_key
- [x] spanner_create_table_with_timestamp_column
- [x] spanner_create_index
- [x] spanner_add_column
- [x] spanner_add_timestamp_column
- [x] spanner_add_numeric_column
- [x] spanner_write_read_empty_int64_array
- [x] spanner_write_read_null_int64_array
- [x] spanner_write_read_int64_array
- [x] spanner_write_read_empty_float64_array
- [x] spanner_write_read_float64_array
- [x] spanner_create_backup
- [x] spanner_create_backup_with_encryption_key
- [x] spanner_restore_backup
- [x] spanner_restore_backup_with_encryption_key
- [x] spanner_cancel_backup_create
- [x] spanner_list_database_operations
- [x] spanner_delete_backup
- [x] spanner_update_backup

### PR DESCRIPTION
**Pull request by NivedhaSenthil**
- use the new wrapper instead of initialising the generated client directly

## Checklist
- [x] **Tests** pass
- [x] **Lint** pass: `bundle exec rubocop`

Closes #891

### CODE REVIEW

**Comment by hengfengli** [comment]
@NivedhaSenthil Can you please fix the CI errors? Also, why do we put a Do Not Merge label for this PR?

**Comment by NivedhaSenthil** [comment]
@hengfengli Thanks for taking a look at this... the changes for the admin wrapper is not released yet.. hence wanted to wait till it is released. Also the test failures are due to the same reason as it is using the latest released version of `google-cloud-spanner` library which don't have the changes yet.

Created a PR to track on the completed samples and get some early feedback.

File: spanner/spanner_samples.rb [hunk 1]
```diff
@@ -113,12 +131,21 @@ def create_database_with_version_retention_period project_id:, instance_id:, dat
...
 
   require "google/cloud/spanner"
+  require "google/cloud/spanner/admin/database"
 
-  spanner  = Google::Cloud::Spanner.new project: project_id
-  instance = spanner.instance instance_id
-  version_retention_period = "7d"
+  database_admin_client = Google::Cloud::Spanner::Admin::Database.database_admin project_id: project_id
+
+  instance_path = database_admin_client.instance_path project: project_id, instance: instance_id
```
Comment (hengfengli) [inline spanner/spanner_samples.rb:138]: So we don't store project_id in `database_admin_client`, right? We have to pass `project_id` again here.

**Review by hengfengli — APPROVED** [review]
LGTM.

File: spanner/spanner_samples.rb (hunk 1 above)
Comment (NivedhaSenthil) [reply spanner/spanner_samples.rb:138]: yes, we can remove it from client creation :+1:

**Comment by NivedhaSenthil** [comment]
Skipping backup tests due to timeout issue. To be fixed as part of https://github.com/GoogleCloudPlatform/ruby-docs-samples/issues/893

**Review by bajajneha27 — APPROVED** [review]
LGTM

**Review by dazuma — APPROVED** [review]

**Pull request merged**